Added --chmod, -a, -n, changed Windows backslashes to slashes, updated README.md

karland opened the pull request:
1. `--chmod` option (especially for running under Windows).
1. Added very common `-a` option
1. Added dry-run option `-n` for making a careful start and testing when starting to use `gulp-rsync`.
1. Added option `command` to log the `rsync`-command that is generated from `rsync.js`.
1. Dealt with path issue. It seems that `rsync` cannot deal with `\\`. After escaping the `\` from Windows paths in `rsync.js` `rsync` did not generate implied directories. Changed `\` into `/`.
1. Updated `README.md` with additional options from above and more details from `rsync` man pages for higher user convenience and transparency.

born2net commented:
can this be merged in?

jerrysu commented on rsync.js:
```diff
@@ -123,6 +123,7 @@ function escapeShellArg(arg) {
   if (!/(["'`\\ ])/.test(arg)) {
     return arg;
   }
+  arg = arg.replace(/([\\])/g, '/');
```
Is this necessary? Is this Windows related?

jerrysu commented:
Thanks for this! I'll likely merge this as it seems to resolve a lot of the issues that people have been having with this plugin. I'm not a Windows user, so I haven't been able to address these issues.

karland commented:
@jerrysu Welcome. To answer your question: Yes it is Windows related. After I implemented my PR changes I got 

```
rsync -avzR ...
```

from `gulp-rsync`. The command was what I wanted, but the syncing did not work as expected. Basically I did not get the implied directories. In comparison plain `rsync` worked perfectly with the same command. So, I wondered where the difference was coming from. The difference was coming from escaping the backslashes into `\\`. As a result `rsync` threw me a warning that it was having difficulties to work with Windows `\` and wanted UNIX '/'. So I added the line in question and then it worked as expected. My hunch is that not escaping the backslashes would also work, because than `rsync` is making the change. But I have not tested that. I am not sure why the escaping of the backslashes is actually necessary `\`.

nojimage commented:
:+1:  I want `-a` and `-n` options!

gaving commented:
:+1:  Any update on this merge? On Windows and encountering permissions issues which this PR sorts.

jaydenseric commented:
Really need the `chmod` option! Can't use this plugin until then...

pyrsmk commented:
The same.

thenationofalex commented:
:+1:  Would be great if a merge happened soon.

karland commented:
@jerrysu Hi, is there anything I can do concerning my PR that keeps you from merging it? If yes, please let me know. Best.

nikini commented:
Is this gonna be merged at some point?
